app/controllers/application_controller.rb

```diff
@@ -1,3 +1,20 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
+
+  helper_method :logged_in?, :current_user
+
+  def logged_in?
+    !!current_user
+  end
+
+  def current_user
+    current ||= User.find(session[:user_id]) if session[:user_id]
```

Comment:
http://railscasts.com/episodes/1-caching-with-instance-variables

This rails cast can actually help you to understand way better than I described :)